Iniciante/1961/README.md

```diff
@@ -0,0 +1,77 @@
+
+Neste jogo, a distância entre os canos é irrelevante, ou seja, o sapo sempre consegue alcançar o próximo cano com um pulo.
+
+Você deve escrever um programa que, dadas as alturas dos canos e a altura do pulo do sapo, mostra se a fase do jogo pode ser vencida ou não.
+
+**Problema completo**: https://www.urionlinejudge.com.br/judge/pt/problems/view/1961
+
+# Resolução
+
+Para esse exercício utilizaremos um conceito de uma variável que funciona como uma chave/interruptor.
+
+Nesse exercício, utilizamos a biblioteca `<stdlib.h>` para fazer uso da função `abs()` que retorna o valor absoluto (positivo) do que for passado como argumento (ex: abs(-3) = 3).
+
+Começaremos declarando nossas variáveis. `pulo` e `canos` armazenarão a entrada do problema, altura do pulo e quantidade de canos, respectivamente. a variável `sobreviveu` nos dirá se o sapo conseguiu chegar ao fim ou não, `i` será nosso indice para ir de um cano ao outro e o [ponteiro](https://www.treinaweb.com.br/blog/ponteiros-em-c-uma-abordagem-basica-e-inicial/) `altura_canos`que em uma breve explicação, se comportará como um vetor, porém, não definimos ainda o tamanho desse vetor, onde iremos armazenar os tamanhos dos canos.
+
+```c
+    int pulo, canos,  sobreviveu = 1, i;
+    int *altura_canos;
+```
+
+Assim, com as variáveis declaradas podemos ler a entrada e com ela, fazer uma [alocação dinamica](http://linguagemc.com.br/alocacao-dinamica-de-memoria-em-c/) que irá reservar na memória um espaço equivalente a o tamanho de uma variável tipo `int` multiplicada pela quantidade de canos que teremos e esse será nosso vetor.
+
+
+```c
+    scanf("%d %d",&pulo,&canos);
+    altura_canos = malloc(sizeof(int)*canos);
+``` 
+
+Com a alocação feita, `altura_canos` se comporta como um vetor, então podemos preenchê-lo com os valores da entrada do problema, percorrendo todas as posições (canos) com um loop `for` de 0 a `canos`. 
+
+```c
+    for(i=0;i<canos;i++){
+        scanf("%d",&altura_canos[i]);
+    }
+```
+
+Assim, podemos fazer as verificações agora. Com um loop `for` que percorre todas as posições, porém com **muito cuidado** já que nossas verificações serão sempre relacionando o cano atual com o próximo, a nossa condição de parada se dá por `i<canos-1` pois se chegassemos no último cano e tentássemos acessar o próximo, chegariamos á um erro, já que não existe próximo cano. 
+
+A condição do `if` diz que, caso o valor absoluto (retornado pela função abs()) seja maior que o tamanho do pulo do sapo, temos a condição em que o mesmo não conseguiria sobreviver, portanto, nossa variável sobreviveu recebe 0. Note que independente do que virá a seguir, essa variável nunca volta a possuir valor 1.
+
+
+```c
+    for(i=0;i<canos-1;i++){
+        if(abs(altura_canos[i] - altura_canos[i+1]) > pulo){
```

Comment:
nao sei se vale a pena botar um && sobreviveu = 1 no condicional do if, por que assim q falhar, encerra o prog sem precisar fazer mais verificações

Reply:
até por que o numero de canos é baixo, 100, fica a seu critério

Reply:
Pra realmente para a execução, precisaria colocar no for e talvez fique confuso (já que não vai ficar algo lento por ser no max 100 canos). no if não vai fazer diferença prática.